libraries/TGRSIAnalysis/TAngularCorrelation/TAngularCorrelation.cxx

```diff
@@ -697,7 +1056,146 @@ std::map<Int_t,std::map<Int_t,Int_t>> TAngularCorrelation::GenerateIndexMap(std:
 
    return indexmap;
 }
+////////////////////////////////////////
+/// Assign Groups for Angular Indexes
+///
+/// \param[in] group vector of assigned groups for angular indexes
+/// \param[in] anglemap
+
+std::vector<Int_t> TAngularCorrelation::AssignGroups(std::vector<Int_t> &group, std::vector<Double_t> &anglemap)
+{
+   std::vector<Int_t> groups; // vector to return
+
+   // get number of group entries
+  Int_t size = group.size();
+
+// basic consistency check
+  if (size!=(Int_t) anglemap.size()) {     
+ } 
+   
+   // initialize group vector
+   for (Int_t i=0;i<size;i++) {
+      groups.push_back(group[i]);
+   }
+
+   return groups;
+}
+////////////////////////////////////////
+/// Assign average angles for groups
```

Comment:
I think "average" is the wrong word to use here - I think "Assign angles for groups" will work just fine.